src/EditorFeatures/Test/Diagnostics/DiagnosticAnalyzerServiceTests.cs

```diff
@@ -961,40 +902,32 @@ void M()
                     else
                         workspace.OpenDocument(document.Id);
 
-                    await incrementalAnalyzer.AnalyzeDocumentAsync(document, bodyOpt: null, InvocationReasons.SemanticChanged, CancellationToken.None);
                     break;
 
                 case BackgroundAnalysisScope.FullSolution:
-                    await incrementalAnalyzer.AnalyzeProjectAsync(project, semanticsChanged: true, InvocationReasons.Reanalyze, CancellationToken.None);
                     break;
             }
 
+            await incrementalAnalyzer.ForceAnalyzeProjectAsync(project, CancellationToken.None);
             await ((AsynchronousOperationListener)service.Listener).ExpeditedWaitAsync();
 
             var root = await document.GetSyntaxRootAsync();
             text = await document.GetTextAsync();
-            if (analysisScope == BackgroundAnalysisScope.None)
```

Comment:
since we're just 'pulling' on the reuslts now, these sorts of flag have no meaning in the part of the system that computes results.  Instead, it is up to the "puller" (like our LSP server) to encode this logic.  Since the tests are just asking for the computed data, they get the results.